sound/soc/sof/ops.h

```diff
@@ -305,43 +305,37 @@ static inline int snd_sof_debugfs_add_region_item(struct snd_sof_dev *sdev,
 static inline void snd_sof_dsp_write(struct snd_sof_dev *sdev, u32 bar,
 				     u32 offset, u32 value)
 {
-	if (sof_ops(sdev)->write) {
+	if (sof_ops(sdev)->write)
 		sof_ops(sdev)->write(sdev, sdev->bar[bar] + offset, value);
-		return;
-	}
-
-	dev_err_ratelimited(sdev->dev, "error: %s not defined\n", __func__);
+	else
+		writel(value,  sdev->bar[bar] + offset);
```

Comment:
Would it be more explicit to fall back to the SOF variants of `sof_io_*`? It will make it easier to spot if the given arch does not need to set up these callbacks?

Reply:
well, we can't remove those sof_io* routines since they are used by AMD and I am not touching their code. So the answer is likely to fallback on the sof_io wrapper. Thanks for the comment @ujfalusi

Reply:
> well, we can't remove those sof_io* routines since they are used by AMD and I am not touching their code. So the answer is likely to fallback on the sof_io wrapper. Thanks for the comment @ujfalusi

Directly? They should be using the wrapper at all times.

Reply:
readb, readl and readq are recommended by Linux driver development. Now we already have wrapper for them like snd_sof_dsp_xxx, so it is applicable.